tests/components/slide_local/test_config_flow.py

```diff
@@ -0,0 +1,346 @@
+            CONF_PASSWORD: "pwd",
+        },
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"]["base"] == error
+
+    # tests with all provided
+    mock_slide_api.slide_info.side_effect = None
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: HOST,
+            CONF_PASSWORD: "pwd",
+        },
+    )
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == HOST
+    assert result["data"][CONF_HOST] == HOST
+    assert result["data"][CONF_PASSWORD] == "pwd"
+    assert result["data"][CONF_API_VERSION] == 2
+    assert not result["options"][CONF_INVERT_POSITION]
+
+
+async def test_abort_if_already_setup(hass: HomeAssistant) -> None:
+    """Test we abort if the device is already setup."""
+
+    MockConfigEntry(domain=DOMAIN, data={CONF_HOST: HOST}).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: HOST,
+            CONF_PASSWORD: "pwd",
+        },
+    )
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_zeroconf(
+    hass: HomeAssistant, mock_slide_api: AsyncMock, mock_setup_entry: AsyncMock
+) -> None:
+    """Test starting a flow from discovery."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_ZEROCONF}, data=MOCK_ZEROCONF_DATA
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "zeroconf_confirm"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={}
+    )
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == "127.0.0.2"
+    assert result["data"][CONF_HOST] == "127.0.0.2"
+    assert not result["options"][CONF_INVERT_POSITION]
+    assert result["result"].unique_id == "12:34:56:78:90:ab"
+
+
+async def test_zeroconf_duplicate_entry(
+    hass: HomeAssistant, mock_slide_api: AsyncMock, mock_setup_entry: AsyncMock
+) -> None:
+    """Test starting a flow from discovery."""
+
+    MockConfigEntry(
+        domain=DOMAIN, data={CONF_HOST: HOST}, unique_id="12:34:56:78:90:ab"
+    ).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_ZEROCONF}, data=MOCK_ZEROCONF_DATA
+    )
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+    entries = hass.config_entries.async_entries(DOMAIN)
+    assert entries[0].data[CONF_HOST] == "127.0.0.2"
```

Comment:
Would be fun if we can also have a test (or have it here) that we do it with a different IP adres and then assert that it gets updated